docs/FinalReport.md

```diff
@@ -0,0 +1,86 @@
+<https://github.com/JohnStarich/ee461l>
+
+## UI Repo
+<https://github.com/JohnStarich/ee461l-ui>
+
+## Motivation
+
+There's no easy way to find movies groups of people want to watch. Our audience doesn't have a good way to choose movies they'd like to see, especially not with groups with various preferences.
+
+## User Benefits
+
+Users can register on the site and browse the movie database. Users would spend a lot less time trying to brainstorm different movies to watch and they would spend more time watching movies. Our web application is very easy to navigate, add friends, create groups, and it is not distracting. With a click of a button, a user’s group will be provided with the movies that the group would benefit largely from.
+
+## Feature Description and Requirements
+
+1. Users can register, login and rate movies they've seen.
+2. Users can link up with their friends and generate a "Top 20" list of movies that they will most likely enjoy together. 
+3. Users can change their name and password. 
+4. Users are able to add and delete friends from their friends list, and from their friends list create groups.
+
+## Design
+
+![Design](screenshots/design.png)
+
+![Updated Class Diagram](screenshots/classDiagram1.png)
+
+![Updated Class Diagram](screenshots/classDiagram2.png)
+
+## User Interface
+
+![Login](screenshots/login.png)
+
+![Register](screenshots/register.png)
+
+![Movie Search w/ Rating](screenshots/movie_search_wrating.png)
+
+![Friends List](screenshots/friendslist.png)
+
+![Find Friends](screenshots/findfriends.png)
+
+![Add Friends](screenshots/addfriends.png)
+
+![Make Group](screenshots/makegroup.png)
+
+![Add Group](screenshots/addedgroup.png)
+
+![Add friends to group](screenshots/addfriendstogroup.png)
+
+![Group has members](screenshots/grouphasfriends.png)
+
+![Group Recommendations](screenshots/grouprecommendations.png)
+
+![Settings](screenshots/settings.png)
+
+![Change Name](screenshots/changename.png)
+
+## Testing Tools, Test Scenarios, and Test Cases
+
+Our application was divided into two different parts for testing, the first is the API (back end) and the second is the UI (front end). The team’s plan was to make sure that with each pull request on git hub our systems would be tested. 
+For each class within the API a JUnit test was created, this way we could test the functionality of each class separately. The Rating class was tested to make sure that a user could easily rate any movie, the loading, saving, and modifying aspects were also tested. The User class was tested so that each User could add friends, delete friends, create groups, delete groups, and login. The Session class was mainly tested to make sure that a session would be deleted from our database after x amount of time. The Group class was tested to ensure that adding friends into groups was fully functional as well as changing group names. The Movie class was tested to see that the equals method worked well. This completes the unit testing. 
```

Comment:
The Rating class was tested to verify that a user could easily rate any movie. The loading, saving, and modifying aspects of the Rating class were also tested.